### PR TITLE
validate the commstemplate classification against the classificationtype table

### PR DESCRIPTION
**Pull request by jwgwalton**
Speaking to @JamesGardiner about the mocking. Not sure whether it reduces readability or improves.

Setting up the test data for the template unit tests now involves
- create classification type
- create comms template (which is validated  against that classification type)

DO TEST

### CODE REVIEW

**Review by benjefferies — CHANGES_REQUESTED** [review]
Why are we moving away from the classification types enum to the classification types in the DB, it seems like the initial solution was simpler and was good enough?

**Comment by jwgwalton** [comment]
@benjefferies  They want to be able to update the possible classifiers going forward. I put in the previous data model for development. Theres a link to the design document somewhere ( I don't think you or I have access to the confluence page)

**Comment by benjefferies** [comment]
Happy to approve this based on discussions. Going forward it would be worth questioning the initial design. We definitely don't need any functionality to update classification types and the moment and we don't have a requirement to yet so it may have been simpler to keep it as an enum until we had a requirement to do anything more complex

**Review by benjefferies — APPROVED** [review]

File: application/controllers/template_controller.py [hunk 1]
```diff
@@ -34,14 +35,26 @@ def get_templates_by_classifiers(classifiers):
...
-def _validate_template(template):
+def _validate_template_schema(template):
     try:
         validate(template, template_schema)
     except ValidationError as exception:
         logger.exception('Attempted to upload invalid template')
         raise InvalidTemplateException(exception.message, status_code=400)
 
 
+def _validate_classification_types(template_classifications):
```
Comment (benjefferies) [inline application/controllers/template_controller.py]: The name may be more readable as `_validate_classification_types_exist`

**Pull request merged**